app/models/ability.rb

```diff
@@ -0,0 +1,19 @@
+class Ability
+  include CanCan::Ability
+
+  class Ability
+    include CanCan::Ability
+
+    def initialize(user)
+      can :read, Post, public: true
+
+      return unless user.present? # additional permissions for logged in users (they can read their own posts)
+
+      can :read, Post, user: user
+
+      return unless user.admin? # additional permissions for administrators
+
+      can :read, Post
+    end
+  end
```

Comment:
- [ ] As per the project [requirments](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_add_authorization.md);

- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). 
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). 
Please kindly implement the above requirement, something like so:

 class Ability
 include CanCan::Ability

     def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
     end
 end